frontend/src/scenes/notebooks/Notebook/migrations/migrate.ts

```diff
@@ -0,0 +1,37 @@
+import { JSONContent } from '@tiptap/core'
+import { NodeKind } from '~/queries/schema'
+import { NotebookNodeType, NotebookType } from '~/types'
+
+// NOTE: Increment this number when you add a new content migration
+// It will bust the cache on the localContent in the notebookLogic
+// so that the latest content will fall back to the remote content which
+// is filtered through the migrate function below that ensures integrity
+export const NOTEBOOKS_VERSION = '1'
+
+export function migrate(notebook: NotebookType): NotebookType {
+    let content = notebook.content.content
+
+    if (!content) {
+        return notebook
+    }
+
+    content = convertInsightToQueryNode(content)
+    return { ...notebook, content: { type: 'doc', content: content } }
+}
+
+function convertInsightToQueryNode(content: JSONContent[]): JSONContent[] {
+    return content.map((node) => {
+        if (node.type != 'ph-insight') {
+            return node
+        }
+
+        return {
+            ...node,
+            type: NotebookNodeType.Query,
+            attrs: {
```

Comment:
I think we need to copy `title` too

Reply:
and height?

Reply:
I specifically left them out because I wanted to fall back to the defaults in the query node

Reply:
Ah, it meant for me a notebook that had the saved insight title for multiple insights now said "HogQL" for the title of everything

Reply:
Ah, that's because the correct title logic is in a different PR 🤦 

https://github.com/PostHog/posthog/pull/17289/files#diff-c1c303fd13bcb91112ea40b294c4fd493f0e6c7085d5c481544aac3fee8ec2b0R102

Would it be worth merging this PR into the linked PR or just merging them both at the same time. The title should only be changed when a customer opens the notebook (fwiw there is a very small number of affected notebooks)

Reply:
No need to make it complicated. You're on it 🙌